src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/ReadPropertiesStepExecution.java

```diff
@@ -49,17 +52,17 @@
 
     private static final long serialVersionUID = 1L;
 
-    @StepContextParameter
-    private transient TaskListener listener;
-
-    @Inject
-    private transient ReadPropertiesStep step;
+    protected ReadPropertiesStepExecution(@Nonnull ReadPropertiesStep step, @Nonnull StepContext context) {
+        super(step, context);
+    }
 
     @Override
     protected Map<String, Object> doRun() throws Exception {
         PrintStream logger = listener.getLogger();
+        ReadPropertiesStep step = (ReadPropertiesStep)getStep();
```

Comment:
A little ugly—why not keep the `transient ReadPropertiesStep step` field here, rather than have the superclass pretend to manage it but then require casts?